docs/Practices/hub/examples.md

```diff
@@ -0,0 +1,113 @@
+# Examples
+
+## Hub Directory Structure
+```
+// WIP
+bounded-contexts
+  some-bc-1
+    domain            // base domain code for some-bc-1
+    data              // integration code for external resources
+    dependent-bcs     // integration code for other bounded contexts
+    legacy-handlers   // old code not following new architecture
+    repository        // shared database code for all slices
+      crud-fns        // functions that mutate data a single entity w/o condition or search by a unique id w/o filtering
+        find-entity1-by-id.ts
+        insert-entity2.ts
+      entity1.ts      // types representing database tables required for entity1
+      entity2.ts      // types representing database tables required for entity2
+    commands
+      some-command-1
+        index.ts      // An index.ts file with API, command, domain, and repository regions
+      some-command-2
+        api.ts
+        command.ts
+        domain.ts
+        repository.ts
+      some-command-3
+        handlers
+          request.ts
+          event.ts
+        command.ts
+        domain.ts
+        repository.ts
+    queries
+      some-query-1
+        index.ts      // An index.ts file with API, query, domain, and repository regions
+      some-query-2
+        api.ts
+        domain.ts
+        query.ts
+        repository.ts
+    routes            // Routes go in this dir, not much guidelines past this. Reference routing point in [the future](hub.md#the-future) for possible iterations
+      router1.ts
+      router2.ts
+  some-bc-2
+```
+
+## Endeavour to make invalid states impossible
+
+### Example 1
+If performing operations that utilize an email address which must be validated, consider creating a branded type rather than utilizing a `string`. This way it becomes very clear which functions within your application need a validated email and which ones don't care.
+```typescript
+// Original
+const sendEmail = (emailAddress: string) => {/* stuff */}
+
+// With Branded Type
+type ValidatedEmail = { 
+  _tag: "ValidatedEmail"; 
+  value: string 
+}
+const sendEmail = (emailAddress: ValidatedEmail) => {/* stuff */}
+``` 
+
+### Example 2
+Sometimes you need to represent a single concept which can take on different shapes based on it's kind. It can be tempting to tack on all the fields that might be required for all the kinds on a single type. Here are a few of the many problems you can face by doing this:
```

Comment:
```suggestion
Sometimes you need to represent a single concept which can take on different shapes based on its kind. It can be tempting to tack on all the fields that might be required for all the kinds on a single type. Here are a few of the many problems you can face by doing this:
```